lib/applyToDefaults.js

```diff
@@ -73,7 +73,7 @@ internals.applyToDefaultsWithShallow = function (defaults, source, options) {
 internals.reachCopy = function (dst, src, path) {
 
     for (const segment of path) {
-        if (!(segment in src)) {
+        if (src[segment] === undefined) {
```

Comment:
Is it possible to check if `src === undefined` a couple lines down and return if so?

I'm not sure because I didn't write this code, but it seems like this change (as it is currently written) could have implications with triggering `Proxy` objects.

Reply:
I'm not sure I understand... do people use proxy objects with an expectation that `a in b` is used internally in hoek? If so, that seems like they depend on a private behaviour.

Reply:
Sorry for the confusion. What I meant is that `Proxy` traps can have side effects, so we should try to avoid triggering them when possible:

```js
'use strict';
const target = { foo: 0 };
const handler = {
  get(target, prop, receiver) {
    target.foo++;
    return target.foo;
  }
};
const proxy = new Proxy(target, handler);

console.log('bar' in proxy);
console.log(target.foo);

console.log(proxy.bar);
console.log(target.foo);
```

In this example, there are no side effects (changes to `target.foo`) from `'bar' in proxy`, but there are from `proxy.bar`.

Reply:
Fixed 👍